### ISSUE TITLE
Note: Tesla API access now seems to be blocked on various cloud providers

### ISSUE THREAD

**Issue by MattJeanes** [adriankumpf/teslamate#914]
As of last night, myself and others have been reporting Tesla API timeouts that only seem to occur from cloud providers such as Azure in my case. Looks like Tesla may have enforced some kind of ip blacklist. I'm just creating this issue as a heads up as people are likely to come here wondering why their TeslaMate instance is no longer working.

Personally I have moved my TeslaMate to a local raspberry pi, keeping the database / grafana on Azure and that works again for me. Hope this issue helps any fellow confused TeslaMate users

You might want to add some kind of warning to your site / README.md about this - I hope it's a temporary block but we have no way of knowing

**Comment by GtHbbrr** [comment]
Thanks! How do you run TM on pi and keep the database in the cloud instance?

**Comment by MattJeanes** [comment]
There are a few different ways, but personally, I added a port mapping `5432:5432` to the database container in docker-compose file to essentially expose PostgreSQL to the host network, be very careful when doing this and ensure you have a firewall in place on the server whitelisted to your home IP address (or wherever the pi is) otherwise anyone can see and get into your database and see all your TeslaMate information which is quite sensitive e.g. locations etc 

Then I basically moved the TeslaMate part of the docker-compose out, deleted that particular container on the cloud server, and put it into a new one running on the pi, then all I had to do was change DATABASE_HOST to the cloud server name, and it was all working again.

Please note that this is more dangerous than running everything locally on a pi as you are exposing your database to attacks from the public internet unless you add a ip whitelist to your home ip, hopefully you have a static home ip address!

**Comment by Niek** [comment]
FWIW: DigitalOcean IP ranges still seems to be fine.

**Comment by gundalow** [comment]
@adriankumpf I wonder if it might be worth pinning this issue in GitHub?

If someone could give me details of what the error message or symptoms are when hosting teslamate on an IP range that's blocked I'll happily update the docs.

**Comment by cowlinb6** [comment]
FYI, this looks to be affecting more than TeslaMate.

![Screenshot 2020-09-11 at 13 55 50](https://user-images.githubusercontent.com/12410874/92928130-9432f400-f436-11ea-8556-115b8d412ec2.png)

**Comment by jonatannietoa** [comment]
We are having problems with Tesla Mate in AWS, and we are a big Spanish community with Tesla Mate hosted in AWS!

**Comment by GtHbbrr** [comment]
Interesting. How large is your community and what is your use case?

**Comment by jonatannietoa** [comment]
We have +1000 people doing our tutorials and installing Tesla Mate in AWS. We do not have how many people have installed Tesla Mate in AWS, but we are receiving messages about it.

**Comment by GtHbbrr** [comment]
Wow 1000+ people. And the number of users is a multiple of that?

**Comment by jonatannietoa** [comment]
No, not everyone has installed it, but 50+ have Tesla Mate in AWS. We have a tutorial and script for unexperienced users: https://github.com/jonatannietoa/tesla-mate

They only need change passwords.

**Comment by GtHbbrr** [comment]
Have you tried contacting the UK Teslamate group? They are also very active using the instructions from Dave at TeslaEV.co.uk

**Comment by GtHbbrr** [comment]
I am now installing teslamate in another cloud at Digital Ocean. Let me know if you need help with this

**Comment by lmediero** [comment]
I'm having the same issue in Oracle Cloud, the teslamate server lost the connection with the API the 9th September.

Any idea about how to solve it?

**Comment by merlinchen** [comment]
Seems only the websocket api is down?

**Comment by lmediero** [comment]
> Seems only the websocket api is down?

Could be, but it strange that only happens from cloud providers (Azure, WAS, OCI and GCS).

**Comment by 0ff** [comment]
Is there any option to have teslamate use a Proxy for API Calls? I am also having problems with an instance running on AWS but I don't want to move the actual instance out of AWS. Instead, I'd like to configure it such that it uses a proxy running on a non-blocked server that would then just forward the requests for the AWS instance.

If I read the code correctly, then this is now impossible due to strict certificate checks and no option to configure a (SOCKS, HTTPS, ...) proxy. Should I open a new ticket for this request?

**Comment by Saxifragus** [comment]
> Then I basically moved the TeslaMate part of the docker-compose out, deleted that particular container on the cloud server, and put it into a new one running on the pi, then all I had to do was change `DATABASE_HOST` to the cloud server name, and it was all working again.

@MattJeanes can you share with us the contents of the "local" docker-compose.yml file? I have trouble understanding how I can connect to a database instance in the cloud.

**Comment by 0ff** [comment]
@Saxifragus for me I'm now doing the same thing (rather than proxying the requests) with the following local docker-compose.yml:
```yaml
version: "2"
services:
  sshtunnel:
    image: 0fff/alpine-ssh-tunnel:latest
    restart: always
    environment:
      REMOTE_HOST: my-instance-name.eu-west-1.rds.amazonaws.com
      REMOTE_PORT: 5432
      LOCAL_PORT: 5432
      SSH_USER: user
      SSH_HOST: my.bastion.server.at.aws.com
      SSH_PORT: 22
    volumes:
      - ./development.pem:/root/.ssh/id_rsa
    networks:
      - tunnelnet
    ports:
      - "5432:5432"
  teslamate:
    image: teslamate/teslamate:edge
    restart: always
    environment:
      DATABASE_USER: my-database-user
      DATABASE_PASS: my-database-pass
      DATABASE_NAME: my-database-name
      DATABASE_HOST: db
      DISABLE_MQTT: "true"
      VIRTUAL_HOST: teslamate.x.y.z
      TZ: Europe/Berlin
    cap_drop:
      - ALL
    stdin_open: true
    links:
      - sshtunnel:db
    depends_on:
      - sshtunnel
    networks:
      - tunnelnet
networks:
  tunnelnet:
```

So I'm using a host at aws as a ssh proxy to my rds instance that is not publicly accessible. This whole setup is running on my pi and requests work fine.

**Comment by MattJeanes** [comment]
@Saxifragus Sure, here is the docker-compose (sensitive information masked of course) from my Raspberry Pi

This section was essentially moved out of the docker-compose on the cloud server with the only difference being the `DATABASE_HOST`

```yaml
version: "3"

services:
  teslamate:
    image: teslamate/teslamate
    container_name: teslamate
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=**redacted**
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=myserver.cloudprovider.net # (or an ip address)
      - DISABLE_MQTT=true
      - TZ=Europe/London
      - LOCALE=en
    cap_drop:
      - all
    ports:
      - 4000:4000
    restart: always
```

Meanwhile in the server I exposed the PostgreSQL database to the host and firewalled it appropriately to stop attackers on the cloud provider portal:

```yaml
  postgres:
    image: postgres
    container_name: postgres
    networks:
      network:
        aliases:
        - "postgres.internal"
    ports:
      - 5432:5432 # this is the new bit
    environment:
      - POSTGRES_PASSWORD=**redacted**
    volumes:
      - postgres-data:/var/lib/postgresql/data
    restart: always
```

**Comment by Saxifragus** [comment]
Thanks a lot, @0ff and @MattJeanes
I'll try and get this working for me.

**Comment by mikeh2010** [comment]
TezLab claim to have fixed the third party connection issue, and are looking at the upcoming two factor issue...

**Comment by DaveD-W** [comment]
GCP is now working again, not sure on AWS, Azure or any of the others.

**Comment by nickjupp** [comment]
saw this on my Teslafi account over the weekend (I run both Teslamate locally and TeslaFi for now)

_TeslaFi services were interrupted on September 10th for approximately 12 hours due to an issue with Tesla's servers. A temporary solution has been put in place until the issue is resolved._

**Comment by lmediero** [comment]
> GCP is now working again, not sure on AWS, Azure or any of the others.

Oracle Cloud is also working again 👍🏻

**Comment by mikeh2010** [comment]
AWS also working again.
 I imagine the upcoming two factor auth will probably break teslamate...

**Comment by adamfd** [comment]
Also working again on Azure now.  As you say if there’s a change to the auth mechanism coming up, could well break again...  if it is using oauth though should hopefully be easy to adapt.

**Comment by MattJeanes** [comment]
Since everything seems to be working again, including for myself on Azure now - I'll close this issue. Hopefully things continue to work now, although as @adamfd mentioned incoming 2FA will likely require some changes to various apps including TeslaMate

**Comment by ogezue** [comment]
I'm getting a 401 on Google cloud. Is it working for anybody on google platform?

**Comment by DaveD-W** [comment]
> I'm getting a 401 on Google cloud. Is it working for anybody on google platform?

Health check failed for me too :( - Back to Digital Ocean I go! (fingers crossed that still works!)

**Comment by adamfd** [comment]
Indeed - shortly after posting it was working ok on Azure it also started to fail again.  Curious what is going on for them to open it and then block again!

**Comment by MattJeanes** [comment]
Well it would seem I prematurely closed this issue then!

I'm also seeing issues on Azure, back to my local Raspberry Pi and working again.

**Comment by mikeh2010** [comment]
AWS back to not working...

**Comment by ogezue** [comment]
What is the current state? Is there anything we can do at the moment? I don't want to run my own hardware, cloud is a must.

**Comment by myi2000** [comment]
Is there a way to specify proxy?

**Comment by MostDefiantly** [comment]
I know all the big hosted API scrapers (TeslaFi, etc) were hit as well then rolled out a "fix" in short order. Any idea what their fix was? Was it a Proxy?

**Comment by ctrufan** [comment]
@myi2000 
I believe it uses the "finch" client, which doesn't support proxying.

@MostDefiantly 
Especially given how quickly the fixes were rolled out, I would expect a proxy of some kind (or a VPN tunnel to another service provider, or reconfiguring their cloud to use different networking routing, etc). I would doubt their fix was anything other than "stop hitting it from a blocklisted IP", whatever form that took.

I debated setting up using a different cloud provider, but realistically I'd expect them to block any provider that started getting popular.

**Comment by myi2000** [comment]
@ctrufan 
Thanks for the pointers, yes indeed it is using Finch.  Noticed in Finch doc [](https://hexdocs.pm/finch/Finch.html) there is ability to specify proxy setting to be passed using :conn_opts while defining the :pools (as seen in http.ex).  Not sure how to do specify pool nor it will work even after that due to other issues.

**Comment by e-tplus** [comment]
I found it working again with Google Cloud.

**Comment by MostDefiantly** [comment]
Looks like Azure has started working again too.

**Comment by Jade-sje** [comment]
From October 12th to today, it is still working in Oracle Cloud. 
 
I'm anxious if it won't work again.

**Comment by ogezue** [comment]
still working with Google cloud. Can this issue be closed?
